EveripediaNetwork/ep-ui
# Pwa setup

# PWA Setup

This PR works on setting up PWA for the project

## Observation
- The beforeInstall Prompt needs some requirement to show and it does not show immediately for all devices even after meeting the requirement

<details>
   <summary>Custom Prompt on Android </summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/EveripediaNetwork/ep-ui/assets/67792966/0dd4c97d-42ea-48e0-affd-507a5b5cddb0">

</details>

<details>
   <summary>Custom Prompt on iPhone </summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/user-attachments/assets/d5037975-6984-4355-a5cc-71549378a35d">

</details>


## How to test
- Prompt only shows for mobile screens
- it only shows once when you just open the app and not every time
- clear cache to see prompt again

## How the Splash Screen looks on android Phones

<details>
   <summary>Installed through Google Pixel Chrome Browser</summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/user-attachments/assets/97cd1d25-d8a4-478a-95f9-ff95bb66c622">

</details>

<details>
   <summary>Installed through Samsung Browser</summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/user-attachments/assets/e1ee8290-88cb-4d29-9777-b0e5d9c880c1">

</details>

<details>
   <summary>Installed through Samsung Chrome Browser</summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/user-attachments/assets/823ecd8a-b7b1-482a-b0c8-def073d88230">

</details>

## How the Home screen looks like on Android and iPhones

<details>
   <summary>Android Homescreen</summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/user-attachments/assets/43f3e01f-2771-4d2d-a204-e639e96916df">

</details>

<details>
   <summary>iPhone Homescreen</summary>
 
<img width="611" alt="Screenshot 2024-07-10 at 1 46 11 PM" src="https://github.com/user-attachments/assets/6faa8072-b869-4633-9908-898a7fa6ee6c">

</details>


## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/2848
closes https://github.com/EveripediaNetwork/issues/issues/2899
